PR: https://github.com/odoo/odoo/pull/210137

From: 9b46cf69dfcb941fa837acb66236d4207b679687
From: Lopes Marc (loma)
Date: 2025-12-10 06:12:37

Breaking data model changes scores: del:1 + add:0, change matches:
-    scale_mid_label = fields.Char("Scale Middle Label", translate=True)

Total Changes: 29

[IMP] survey: revamp question type select and add ui improvements

This commit changes the appearance of the question type select field in
the survey question from radio buttons to buttons with an icon.
The fa icons are hardcoded directly in the view.

Other UI improvements:
* Hiding the live session section when relevant
* Suggested answers: adding a placeholder
* Short text Answer tab: inlining first two checkboxes
* Question title field: label h1, placeholder in h1, field take the whole width
* Scale type questions: reorder visually, delete neutral label
* Scale type questions: dynamic placeholder label based on scale values
* Description field in main question view

task-4778050

Part-of: odoo/odoo#210137
Related: odoo/enterprise#101055
Related: odoo/upgrade#8955
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/survey/models/survey_question.py
+++ b/addons/survey/models/survey_question.py
@@ -85,11 +85,11 @@ class SurveyQuestion(models.Model):
     # question specific
     page_id = fields.Many2one('survey.question', string='Page', compute="_compute_page_id", store=True)
     question_type = fields.Selection([
-        ('simple_choice', 'Multiple choice: only one answer'),
-        ('multiple_choice', 'Multiple choice: multiple answers allowed'),
-        ('text_box', 'Multiple Lines Text Box'),
-        ('char_box', 'Single Line Text Box'),
-        ('numerical_box', 'Numerical Value'),
+        ('simple_choice', 'Single-Select'),
+        ('multiple_choice', 'Multi-Select'),
+        ('text_box', 'Long Text'),
+        ('char_box', 'Short Text'),
+        ('numerical_box', 'Number'),
         ('scale', 'Scale'),
         ('date', 'Date'),
         ('datetime', 'Datetime'),
@@ -111,7 +111,7 @@ class SurveyQuestion(models.Model):
         "Save as user email", compute='_compute_save_as_email', readonly=False, store=True, copy=True,
         help="If checked, this option will save the user's answer as its email address.")
     save_as_nickname = fields.Boolean(
-        "Save as user nickname", compute='_compute_save_as_nickname', readonly=False, store=True, copy=True,
+        "Save as nickname", compute='_compute_save_as_nickname', readonly=False, store=True, copy=True,
         help="If checked, this option will save the user's answer as its nickname.")
     # -- simple choice / multiple choice / matrix
     suggested_answer_ids = fields.One2many(
@@ -128,8 +128,13 @@ class SurveyQuestion(models.Model):
     scale_min = fields.Integer("Scale Minimum Value", default=0)
     scale_max = fields.Integer("Scale Maximum Value", default=10)
     scale_min_label = fields.Char("Scale Minimum Label", translate=True)
-    scale_mid_label = fields.Char("Scale Middle Label", translate=True)
     scale_max_label = fields.Char("Scale Maximum Label", translate=True)
+    scale_min_label_placeholder = fields.Char(
+        compute='_compute_scale_min_label_placeholder',
+        string="Min Label Placeholder")
+    scale_max_label_placeholder = fields.Char(
+        compute='_compute_scale_max_label_placeholder',
+        string="Max Label Placeholder")
     # -- display & timing options
     is_time_limited = fields.Boolean("The question is limited in time",
         help="Currently only supported for live sessions.")
@@ -254,6 +259,16 @@ class SurveyQuestion(models.Model):
                     or not question.question_placeholder:  # avoid CacheMiss errors
                 question.question_placeholder = False
 
+    @api.depends('scale_min')
+    def _compute_scale_min_label_placeholder(self):
+        for question in self:
+            question.scale_min_label_placeholder = _("Label for %s", question.scale_min)
+
+    @api.depends('scale_max')
+    def _compute_scale_max_label_placeholder(self):
+        for question in self:
+            question.scale_max_label_placeholder = _("Label for %s", question.scale_max)
+
     @api.depends('is_page')
     def _compute_background_image(self):
         """ Background image is only available on sections. """
